pkg/controller/datavolume-controller.go

```diff
@@ -830,6 +830,18 @@ func (r *DatavolumeReconciler) reconcileDataVolumeStatus(dataVolume *cdiv1.DataV
 
 	curPhase := dataVolumeCopy.Status.Phase
 	if pvc != nil {
```

Comment:
This might be the wrong place to put it, as the function is implying we should only update the status. the regular Reconcile seems like an OK place.

Perhaps we can split this off into its own function (getVddkAnnotations or something).

Reply:
Yes I agree with Maya here. We are updating the annotations on the DataVolume, which should happen in the regular reconcile loop. In general the flow of a loop is:
1. Do something to the object you are reconciling (like adding annotations)
2. Report any observed changes in the status.

Reply:
I moved this out of reconcileDataVolumeStatus and into Reconcile. I did not see a clear path to an Update like there was in reconcileDataVolumeStatus, so I added one.